docs: move changelog to separate file

jorsol opened the pull request:
Move the section of Changelog in README.md file to a CHANGELOG.md file,
follow a format based on Keep a Changelog.

It start from 42.0.0 since is the first version to folow a
semantic versioning.

[skip ci]

sehrope commented:
+1

I'd suggest adding a link to the new file in the README.

jorsol commented:
@sehrope, added link to the new file in README.md

jorsol commented:
Another suggestion?

vlsi commented on CHANGELOG.md:
```diff
@@ -0,0 +1,71 @@
+# Changelog
+All notable changes to this project will be documented in this file.
```
I'm afraid the wording is not quite right.
As far as I understand, https://jdbc.postgresql.org/ would still be the master source for the project. GitHub readme and changelog are the secondary sources for convenience purposes.

jorsol replied:
The wording is from http://keepachangelog.com, I can move the wording of "Older releases" section, to the top and remove the default wording if you like.

Two "changelog" headers? do you mean the header on the middle of README.md and the header on the start of CHANGELOG.md?
I can remove all the reference of changelog in the README.md to keep things simple.

I see the https://jdbc.postgresql.org/ source as a historical point of view, more than a "master source", and in many ways, the CHANGELOG.md can be used as a source that feeds https://jdbc.postgresql.org/, the Unreleased section can be used to include changes in a PR, so when a release is ready, the changelog will be ready at the same time.

The point is to have a human readable list of changes, so if you think that the wording has to be changes let me know.

jorsol commented:
I think this is ready, removed the section from README.md, use different wording for the CHANGELOG.md and update the posts of the website to keep consistency.

I also added the relevant notes of commits applied to the date in the **Unreleased** section.

Please review.

jorsol commented:
Comments? This should be ready to merge.

vlsi commented:
@jorsol , would you please adjust `_posts/*`, would you please adjust https://github.com/pgjdbc/pgjdbc/blob/master/release_notes.sh#L30-L44 template as well?

jorsol commented:
@vlsi sure, let me check.

jorsol commented:
@vlsi done, #981 now the CHANGELOG.md Unreleased notes serves as master for the _posts file.
